sdk/core/azure-core/inc/azure/core/internal/io/file_helpers.hpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ * @brief Internal utility functions for files and directories.
+ *
+ */
+#pragma once
+
+#include <azure/core/platform.hpp>
+
+#include <stdexcept>
+#include <string>
+
+#include <sys/stat.h> // for stat() used to check file size and mkdir() used to create directories
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+#include <windows.h>
+#elif defined(AZ_PLATFORM_LINUX)
+#include <unistd.h>
+
+#include <sys/types.h>
+#endif
+#include <string.h>
+
+namespace Azure { namespace Core { namespace IO { namespace _internal {
+
+  /**
+   * @brief Add basic filesystem functionality.
+   */
+  class FileHelpers final {
```

Comment:
```suggestion
  class FileHelpers final {
    ~FileHelpers() = delete;
```
No destructor - no object of this class can be created, `auto fh = new FileHelpers();` won't compile, i.e. we make this more clearly a "static class".

Reply:
I followed the pattern from `strings.hpp`. Should we do something similar there?

Reply:
IMHO, it's not a bad idea to delete the destructor on a static-only class.

I'm not really concerned about the consequences of someone creating an instance of a class with no methods and no member fields, but it does fall into the category of "helping the user to avoid a potential future footgun."